tensorzero-core/src/db/postgres/mod.rs

```diff
@@ -56,6 +66,37 @@ impl PostgresConnectionInfo {
     }
 }
 
+#[async_trait]
+impl HealthCheckable for PostgresConnectionInfo {
+    async fn health(&self) -> Result<(), Error> {
+        match self {
+            Self::Disabled => Ok(()),
+            Self::Mock { healthy } => {
+                if *healthy {
+                    Ok(())
+                } else {
+                    Err(Error::new(ErrorDetails::PostgresConnection { message: "Unhealthy mock postgres connection".to_string() }))
+                }
+            },
+            Self::Enabled { pool } => {
+                let check = async {
+                    let _result = sqlx::query("SELECT 1")
+                    .fetch_one(pool)
+                    .await
+                    .map_err(|_e| Error::new(ErrorDetails::PostgresConnection { message: _e.to_string() }))?;
+
+                    Ok(())
+                };
+                // TODO(shuyang): customize postgres timeout
+                match timeout(Duration::from_millis(500), check).await {
```

Comment:
Is there a standard sqlx timeout configuration / does it respect that? I'm not fully sure if this additional layer of timeout is really necessary here.